src/spoke/interfaces/IVaultSpoke.sol

```diff
@@ -0,0 +1,113 @@
+// SPDX-License-Identifier: UNLICENSED
+// Copyright (c) 2025 Aave Labs
+pragma solidity ^0.8.0;
+
+import {IERC4626} from 'src/dependencies/openzeppelin/IERC4626.sol';
+import {IERC2612} from 'src/dependencies/openzeppelin/IERC2612.sol';
+import {EIP712Types} from 'src/libraries/types/EIP712Types.sol';
+import {INoncesKeyed} from 'src/interfaces/INoncesKeyed.sol';
+
+/// @title IVaultSpoke
+/// @author Aave Labs
+interface IVaultSpoke is IERC4626, IERC2612, INoncesKeyed {
+  /// @notice Thrown when the given signature is invalid.
+  error InvalidSignature();
+
+  /// @notice Thrown when the maximum deposit limit is exceeded.
+  error MaxDepositExceeded(uint256 maxDeposit, uint256 requestedAssets);
```

Comment:
do we need `requestedAssets`? is it not what the user passes to the fn  so already known?
or thinking more about integrations where the input amount is not completely clear